orderer/consensus/follower/follower_chain.go

```diff
@@ -51,16 +143,128 @@ func (*Chain) Errored() <-chan struct{} {
 }
 
 func (c *Chain) Start() {
-	//TODO skeleton
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.started || c.stopped {
+		c.logger.Debugf("Not starting because: started=%v, stopped=%v", c.started, c.stopped)
+		return
+	}
+
+	c.started = true
+
+	go c.run()
+
+	c.logger.Info("Started")
 }
 
 func (c *Chain) Halt() {
-	//TODO skeleton
+	c.halt()
+
+	select {
+	case <-c.doneChan:
+	}
+}
+
+func (c *Chain) halt() {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.stopped {
+		c.logger.Debug("Already stopped")
+		return
+	}
+	c.stopped = true
+	close(c.stopChan)
+	c.logger.Info("Stopped")
 }
 
-// StatusReport returns the ClusterRelation & Status
+// StatusReport returns the ClusterRelation & Status.
 func (c *Chain) StatusReport() (types.ClusterRelation, types.Status) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	clusterRelation := types.ClusterRelationFollower
+	if c.joinBlock != nil {
+		if err := c.amIInChannel(c.joinBlock); err == nil {
+			clusterRelation = types.ClusterRelationMember
+		}
+	}
+
 	status := types.StatusActive
-	//TODO if (height is >= join-block.height) return "active"; else return "onboarding"
-	return types.ClusterRelationFollower, status
+	if (c.joinBlock != nil) && (c.joinBlock.Header.Number >= c.support.Height()) {
+		status = types.StatusOnBoarding
+	}
+
+	return clusterRelation, status
+}
+
+func (c *Chain) run() {
+	c.logger.Debug("The follower.Chain puller goroutine is starting")
+	pullRetryTicker := time.NewTicker(500 * time.Millisecond) // TODO move constant to options
```

Comment:
In the next PR you just do time.After so the TODO isn't relevant i guess right?

Reply:
correct, this is completely revised, so the TODO is taken care of in the next PR.